drivers/hmis/spec/models/hmis/hud/project_spec.rb

```diff
@@ -166,9 +171,9 @@ def selected_instance
     end
 
     it 'chooses instance specified by type (type > category)' do
+      expected = create(:hmis_form_instance, role: role, entity: nil, custom_service_type: cst)
```

Comment:
_This_ test on the other hand needs to be updated, it's currently failing with an existing, non-manifesting?, "bug".
- Check out `release-140` and run this test, it succeeds
- But if you swap lines 170 and 171 so that `expected` is not the last instance to be created, now it fails
- Why: Data collection features uses `Instance`'s [`detect_best_instance_for_project` method](https://github.com/greenriver/hmis-warehouse/blob/release-140/drivers/hmis/app/models/hmis/form/instance.rb) to choose the best instance. It is not applying any logic to prioritize type over category for service rules, as far as I can tell. So this [updated_at sort](https://github.com/greenriver/hmis-warehouse/blob/42ceab1b683f195a9ba6e69e6045720b4c64c853/drivers/hmis/app/models/hmis/hud/project.rb#L242) takes precedence, and since in the existing version of the test the `expected` is the last to be created, it has been falsely succeeding.
- Impact: I don't think this has any actual impact outside of the world of spec tests, because when you actually go to create a service you use the [`service_form_definition` query](https://github.com/greenriver/hmis-warehouse/blob/42ceab1b683f195a9ba6e69e6045720b4c64c853/drivers/hmis/app/graphql/types/hmis_schema/query_type.rb#L238) which is correctly using `find_definition_for_service_type`.

Maybe the simple fix is to have the data collection features also use `find_definition_for_service_type` specifically for the SERVICE feature only?

Reply:
Remove irrelevant test. There's no such thing as a "best" instance for services in this context where we don't have a service type